From e003965d26961a1f15e91e7436c98bd80f1b9995 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <dvlasenk@redhat.com>
Date: Tue, 13 Aug 2013 16:12:38 +0200
Subject: [ABRT EL6 PATCH 203/203] abrt-dump-oops: add "Machine Check
 Exception" to the list of watched strings rhbz#812537

Step one in MCE handling is to catch them.

This is a backport of commit b551dc2285c68ffb9883f795be889a840fed846b

Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
Signed-off-by: Jiri Moskovcak <jmoskovc@redhat.com>
---
 src/lib/kernel.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/lib/kernel.c b/src/lib/kernel.c
index b95bbe4..991e8a0 100644
--- a/src/lib/kernel.c
+++ b/src/lib/kernel.c
@@ -242,6 +242,16 @@ next_line:
                 oopsstart = i;
             else if (strstr(curline, "IRQ handler type mismatch"))
                 oopsstart = i;
+            /*
+             * MCE examples for various CPUs/architectures (collected 2013-04):
+             * arch/arc/kernel/traps.c:                        die("Machine Check Exception", regs, address, cause);
+             * arch/x86/kernel/cpu/mcheck/winchip.c:   printk(KERN_EMERG "CPU0: Machine Check Exception.\n");
+             * arch/x86/kernel/cpu/mcheck/p5.c:                "CPU#%d: Machine Check Exception:  0x%8X (type 0x%8X).\n",
+             * arch/x86/kernel/cpu/mcheck/mce.c:       pr_emerg(HW_ERR "CPU %d: Machine Check Exception: %Lx Bank %d: %016Lx\n",
+             * drivers/edac/sb_edac.c:                 printk("CPU %d: Machine Check Exception: %Lx Bank %d: %016Lx\n",
+             */
+            else if (strstr(curline, "Machine Check Exception"))
+                oopsstart = i;
 
             if (oopsstart >= 0)
             {
-- 
1.8.3.1

